mne/decoding/time_gen.py

```diff
@@ -914,3 +953,358 @@ def time_generalization(epochs_list, clf=None, cv=5, scoring="roc_auc",
+
+    clf : object | None
+        An estimator compliant with the scikit-learn API (fit & predict).
+        If None the classifier will be a standard pipeline including
+        StandardScaler and a linear SVM with default parameters.
+
+    predict_type : {'predict', 'predict_proba', 'decision_function'}
+        Indicates the type of prediction:
+            'predict' : generates a categorical estimate of each trial.
+
+            'predict_proba' : generates a probabilistic estimate of each trial.
+
+            'decision_function' : generates a continuous non-probabilistic
+                estimate of each trial.
+        Default: 'predict'
+
+    predict_mode : {'cross-validation', 'mean-prediction'}
+        Indicates how predictions are achieved with regards to the cross-
+        validation procedure:
+
+            ``cross-validation`` : estimates a single prediction per sample
+                based on the unique independent classifier fitted in the
+                cross-validation.
+            ``mean-prediction`` : estimates k predictions per sample, based on
+                each of the k-fold cross-validation classifiers, and average
+                these predictions into a single estimate per sample.
+
+        Default: 'cross-validation'
+
+    n_jobs : int
+        Number of jobs to run in parallel. Defaults to 1.
+
+    Attributes
+    ----------
+    y_train_ : np.ndarray, shape (n_samples,)
+        The categories used for training.
+
+    estimators_ : list of list of sklearn.base.BaseEstimator subclasses.
+        The estimators for each time point and each fold.
+
+    y_pred_ : np.ndarray, shape (n_times, n_epochs, n_prediction_dims)
+        Class labels for samples in X.
+
+    scores_ : np.ndarray, shape (n_times,)
+        Mean accuracy score for every time.
+
+    cv_ : CrossValidation object
+        The actual CrossValidation input depending on y.
+    """
+    def __init__(self, cv=5, clf=None, predict_type='predict',
+                 predict_mode='cross-validation', n_jobs=1):
+        super(TimeDecoding, self).__init__(
+            cv=cv, clf=clf, predict_type=predict_type,
+            predict_mode=predict_mode, n_jobs=n_jobs)
+
+    def __repr__(self):
+        s = ''
+        if hasattr(self, "estimators_"):
+            s += ("fitted, %d time points" % len(self.estimators_))
+        else:
+            s += 'no fit'
+        if hasattr(self, 'y_pred_'):
+            s += (", predicted %d epochs" % self.y_pred_.shape[1])
+        else:
+            s += ", no prediction"
+        if hasattr(self, "estimators_") and hasattr(self, 'scores_'):
+            s += ',\n '
+        else:
+            s += ', '
+        if hasattr(self, 'scores_'):
+            s += "scored"
+            if callable(self.scorer_):
+                s += " (%s)" % (self.scorer_.__name__)
+        else:
+            s += "no score"
+
+        return "<TimeDecoding | %s>" % s
+
+    def fit(self, epochs, y=None, picks=None):
```

Comment:
why do a `picks` here? The user can do their picks and then feed it to the classifier, no?